indexing-hadoop/src/main/java/io/druid/indexer/path/DatasourcePathSpec.java

```diff
@@ -0,0 +1,187 @@
+/*
+* Licensed to Metamarkets Group Inc. (Metamarkets) under one
+* or more contributor license agreements. See the NOTICE file
+* distributed with this work for additional information
+* regarding copyright ownership. Metamarkets licenses this file
+* to you under the Apache License, Version 2.0 (the
+* "License"); you may not use this file except in compliance
+* with the License. You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied. See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+
+package io.druid.indexer.path;
+
+import com.fasterxml.jackson.annotation.JacksonInject;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Function;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+import com.metamx.common.logger.Logger;
+import io.druid.indexer.HadoopDruidIndexerConfig;
+import io.druid.indexer.hadoop.DatasourceIngestionSpec;
+import io.druid.indexer.hadoop.DatasourceInputFormat;
+import io.druid.query.aggregation.AggregatorFactory;
+import io.druid.timeline.DataSegment;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.mapreduce.Job;
+import org.apache.hadoop.mapreduce.lib.input.MultipleInputs;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Set;
+
+public class DatasourcePathSpec implements PathSpec
+{
+  private static final Logger logger = new Logger(DatasourcePathSpec.class);
+
+  //Note that we can't get it "injected" because HadoopDruidIndexerConfig uses a custom injector which can't set
+  //appropriate UsedSegmentLister depending upon whether it is running in the context of standalone hadoop indexer
+  //or the hadoop index task via overlord
+  private static volatile UsedSegmentLister segmentLister;
```

Comment:
@gianm 
or may be a `HadoopReindexTask` with a new spec so that if/elsing is not necessary, that internally converts it to more or less a `HadoopIndexTask` spec but with necessary list of segments in it.